SYCL/ESIMD/api/functional/operators/operator_decrement_and_increment.hpp

```diff
@@ -0,0 +1,283 @@
+
+      if constexpr (TestCaseT::is_increment()) {
+        mutate(ref_data, mutator::For_addition<DataT>(1));
+      } else if constexpr (!TestCaseT::is_increment()) {
+        mutate(ref_data, mutator::For_subtraction<DataT>(1));
+      }
+    } else {
+      static const DataT min = value<DataT>::lowest();
+      static const DataT denorm_min = value<DataT>::denorm_min();
+      static const DataT max = value<DataT>::max();
+      static const DataT zero_point_one = static_cast<DataT>(0.1);
+      static const DataT one_point_zero = 1.0;
+
+      if constexpr (TestCaseT::is_increment()) {
+        ref_data.reserve((NumElems > 1) ? NumElems : 6);
+        ref_data.insert(ref_data.end(),
+                        {zero_point_one, denorm_min, -denorm_min,
+                         -one_point_zero + value<DataT>::pos_ulp(42U),
+                         min + value<DataT>::pos_ulp(42U),
+                         (max - 1) + value<DataT>::neg_ulp(42U)});
+        for (size_t i = ref_data.size(); i < NumElems; ++i) {
+          ref_data.push_back(0.1 * i);
+        }
+      } else if constexpr (!TestCaseT::is_increment()) {
+        ref_data.reserve((NumElems > 1) ? NumElems : 6);
+        ref_data.insert(ref_data.end(),
+                        {zero_point_one, denorm_min, -denorm_min,
+                         -one_point_zero + value<DataT>::neg_ulp(42U),
+                         max + value<DataT>::neg_ulp(42U),
+                         (min + 1) + value<DataT>::pos_ulp(42U)});
+        for (size_t i = ref_data.size(); i < NumElems; ++i) {
+          ref_data.push_back(0.1 * i);
+        }
+      }
+    }
+
+    // If current number of elements is equal to one, then run test with each
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed &= run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed &= run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> source_simd_out(NumElems, allocator);
+    shared_vector<DataT> result_simd_out(NumElems, allocator);
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         allocator);
+
+    queue.submit([&](sycl::handler &cgh) {
+      const DataT *const ref = shared_ref_data.data();
+      DataT *source_simd_data_storage = source_simd_out.data();
+      DataT *result_simd_data_storage = result_simd_out.data();
```

Comment:
Nit, just as suggestion:
```suggestion
      DataT *source_simd_data_ptr = source_simd_out.data();
      DataT *result_simd_data_ptr = result_simd_out.data();
```
it's more like `source_simd_out` is a storage for data, and these ones are the USM pointers to access the storage

Reply:
this comment applied [here](https://github.com/intel/llvm-test-suite/pull/827/commits/7ff74964cc8f28e39cce66778060f321038d13cc#diff-957362298136f7a2e0ab66c1cff4239caca52ea84dab6e6b2728ccc9411ebe0eL228-R244)